src/main/utils/file/downloadFile.ts

```diff
@@ -0,0 +1,52 @@
+import wget from 'wget-improved';
+import path from 'path';
+import { mkdir } from '../../util';
+
+type DownloadFile = (
+  url: string,
+  targetFile: string,
+  onStartCallback: () => void,
+  onProgressCallback: (progress: number) => void,
+  onFinishCallback: () => void
+) => Promise<void>;
+
+/**
+ * Download a file from the given `url` into the `targetFile`.
+ */
+const downloadFile: DownloadFile = async (
+  url,
+  targetFile,
+  onStartCallback,
+  onProgressCallback,
+  onFinishCallback
+) => {
+  const options = {
+    protocol: 'https',
+    path: url,
+    method: 'GET',
+  };
+
+  const dirPath = path.dirname(targetFile);
+
+  mkdir(dirPath);
+
+  return new Promise((resolve, reject) => {
+    const download = wget.download(url, targetFile, options);
+
+    download.on('error', (err: any) => {
+      reject(err);
```

Comment:
In line 25 you are setting the `url` as the `path`. In URLs, the path is whatever comes after the domain name / `.com`. Check this amazing image from Mozilla.

![image](https://user-images.githubusercontent.com/86769131/192918223-6875cb5a-85db-4e77-a7ab-bbafaba6a69d.png)


And on line 34 you just give it the entire `url`. I read the rest of the code and the `downloadFile` function always receives a `domain name + path` in the url parameter. So please remove `path` from `options`

Reply:
I did not consider this. Thank you, I've removed path as an option. 😄 😄